src/base/detail/PieImpl.hh

```diff
@@ -153,22 +159,28 @@ struct PieAssigner<Ownership::value, MemSpace::device>
         static_assert(M2 == MemSpace::host,
                       "Can only assign by value from host to device");
 
-        // Always-on once-per-copy check that the number of elements in the
-        // vector doesn't exceed the range. This is in case of a production run
-        // with lots of cross sections, when the incremental PieBuilder checks
-        // are disabled. See comment in PieSlice class documentation.
-        constexpr std::size_t max_pie_size
-            = std::numeric_limits<pie_size_type>::max();
-        CELER_VALIDATE(source.data.size() <= max_pie_size,
-                       "Pie is too large for PieSlice: "
-                           << source.data.size() << " > " << max_pie_size);
-
         PieStorage<T, Ownership::value, MemSpace::device> result{
             DeviceVector<T>(source.data.size())};
         result.data.copy_to_device({source.data.data(), source.data.size()});
         return result;
     }
 };
+
+//---------------------------------------------------------------------------//
+//! Check that sizes are acceptable when taking references
```

Comment:
```suggestion
//! Check that sizes are acceptable when creating references from values
```
(The code seems to be saying the checks are done when the input are `values` the original text can be misunderstood as the input being `references`